Fix for set_fallback_fonts() when markdown=True

Lucas-C opened the pull request:
As a follow-up of #669,
fix the rendering issue visible in this test reference file:
https://github.com/PyFPDF/fpdf2/blob/89c528d/test/fonts/font_fallback.pdf

gmischler commented on docs/Unicode.md:
```diff
@@ -156,13 +154,13 @@ filename.ttf cannot be embedded due to copyright restrictions."
 # Fallback fonts #
 
-FPDF allows you to specify a list of fonts to be used if any character is not available on the font currently used. Common scenarios are use of special characters like emojis within your text, greek characters in formulas or citations mixing different languages.
+The method [`set_fallback_font()`](fpdf/fpdf.html#fpdf.fpdf.FPDF.set_fallback_font) allows you to specify a list of fonts to be used if any character is not available on the font currently set. When a character doesn’t exist on the current font, `fpdf2` will look if it’s available on the fallback fonts, on the same order the list was provided.
```
typo alert: `set_fallback_fonts()` with a plural-s

Lucas-C replied:
Well spotted, thank you!

gmischler commented on fpdf/fpdf.py:
```diff
@@ -3190,13 +3195,11 @@ def _parse_chars(self, txt):
+        "Returns which fallback font has the requested glyph"
         if not self._fallback_font_ids:
             return None
         for font in self._fallback_font_ids:
```
This just returns the first font found that happens to cover the requested character.
Shouldn't we try to use a "best match" strategy instead?

For example, if the original font is bold, and we have both a regular and a bold fallback available covering it, then whichever happens to be listed first will be picked, even if a better match is theoretically available.

gmischler commented:
> I introduced `ignore_style` in my latest commit on this PR, that should address both of your commens @andersonhc & @gmischler

What other possible selection criteria to find the best replacement font can you think of?

Let's say someone later wants to add the possibility to match grotesk/serif fonts with each other. Is it better to introduce another `match_serifs=False|True|"closest"` argument then, or could both be combined in a general "search_strategy" argument?
(I'm not sure if the serif/sans-serif distinction is listed in the font file, but just take this as a placeholder for a characteristic that can actually be found there.)

andersonhc commented:
> > I introduced `ignore_style` in my latest commit on this PR, that should address both of your commens @andersonhc & @gmischler
> 
> What other possible selection criteria to find the best replacement font can you think of?
> 
> Let's say someone later wants to add the possibility to match grotesk/serif fonts with each other. Is it better to introduce another `match_serifs=False|True|"closest"` argument then, or could both be combined in a general "search_strategy" argument? (I'm not sure if the serif/sans-serif distinction is listed in the font file, but just take this as a placeholder for a characteristic that can actually be found there.)

My idea was preserving the order the user specify the fonts and go with the first match.

gmischler commented:
> My idea was preserving the order the user specify the fonts and go with the first match.

So when you have original text in both eg. regular and bold style, then you don't want to take that distinction into account in the replacement font seleced?

gmischler commented:
> This is why I'm not sure to see the point of a more generic mechanism, as it would only apply hypothetically to future usages.
> But it you have a practical alternative to suggest in that case, I'd be curious to know it 😊

This is not about implementing hypothetical features now.

It's about considering features that might possibly become interesting over time, and trying to avoid API choices tha would make adding things later unnecessarily complicated. There are many aspect to any choice of font, and most of them are probably orthogonal to each other. So I think that using a dedicated argument for style criteria (regular/bold/italic) is ok.

However, I'm not sure if a simple "apply/ignore style" distinction is good enough in the long run. I can think of three possible strategies:

* exact match: Use a font with matching style, drop character if none available.
* best match: Search for a font with matching style, otherwise use a default (probably first match in list).
* first match: Ignore style and use the first match in list.

That would give us an argument like: `select_style="EXACT"|"BEST"|"IGNORE"`
"BEST" shouldn't be complicated to implement. But if you don't want to do that now, simply leave it away, then it can be added later.

Lucas-C commented:
@gmischler I have added a commit to this PR implementing what we discussed and adding 2 extra unit tests.
Could you please review this new version?

In the end I went with a boolean `exact_match` flag, as I really did not see the point of the `IGNORE` enum value.
With this value removed, there was really no much benefit in introducing an enum with only 2 possible values,
so I switched to a simple boolean value, with proper documentation.

Lucas-C commented:
Hi @gmischler, @andersonhc

I'm planning on merging this PR next week.
If need be, we will still be able to amend the code in later PRs 😊
